src/overlays/actors/ovl_En_Skj/z_en_skj.c

```diff
@@ -39,201 +125,1497 @@ static ColliderCylinderInitType1 sCylinderInit = {
+        EnSkj_SetupWaitToShootNeedle(this);
+    } else if (this->multiuseTimer != 0) {
+        yawDistToPlayer =
+            Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, this->unk_2F0, 0);
+        this->actor.world.rot.y = this->actor.shape.rot.y;
+        Math_ApproachF(&this->unk_2F0, 2000.0f, 1.0f, 200.0f);
+
+        pos1.x = 0.0f;
+        pos1.y = 0.0f;
+        pos1.z = -120.0f;
+
+        Matrix_RotateY((this->actor.shape.rot.y / 32768.0f) * M_PI, MTXMODE_NEW);
+        Matrix_MultVec3f(&pos1, &pos2);
+        prevPosX = this->actor.world.pos.x;
+        prevPosZ = this->actor.world.pos.z;
+        if (1) {}
+        this->actor.world.pos.x = this->center.x + pos2.x;
+        this->actor.world.pos.z = this->center.z + pos2.z;
+
+        phi_f14 = sqrtf(SQ(this->actor.world.pos.x - prevPosX) + SQ(this->actor.world.pos.z - prevPosZ));
+        phi_f14 = CLAMP_MAX(phi_f14, 10.0f);
+        phi_f14 /= 10.0f;
+
+        this->skelAnime.playSpeed = (yawDistToPlayer < 0) ? -(1.0f + phi_f14) : (1.0f + phi_f14);
+
+    } else if (func_80AFEDF8(this, globalCtx) != 0) {
+        this->backfilpFlag = 1;
+        EnSkj_Backflip(this);
+    } else {
+        EnSkj_SetupResetFight(this);
+    }
+}
+
+void EnSkj_SetupNeedleRecover(EnSkj* this) {
+    Animation_Reverse(&this->skelAnime);
+    EnSkj_SetupAction(this, SKJ_ACTON_NEEDLE_RECOVER);
+}
+
+void EnSkj_NeedleRecover(EnSkj* this, GlobalContext* globalCtx) {
+    if (this->skelAnime.curFrame == 0.0f) {
+        EnSkj_SetupStand(this);
+    }
+}
+
+void EnSkj_SetupSpawnDeathEffect(EnSkj* this) {
+    this->backfilpFlag = 1;
+    EnSkj_SetupAction(this, SKJ_ACTON_SPAWN_DEATH_EFFECT);
+}
+
+void EnSkj_SpawnDeathEffect(EnSkj* this, GlobalContext* globalCtx) {
+    Vec3f effectPos;
+    Vec3f effectVel;
+    Vec3f effectAccel;
+    u32 phi_v0;
+
+    phi_v0 = this->alpha - 4;
+
+    if (phi_v0 > 255) {
+        phi_v0 = 0;
+    }
+    this->alpha = phi_v0;
+    this->actor.shape.shadowAlpha = phi_v0;
+
+    effectPos.x = Rand_CenteredFloat(30.0f) + this->actor.world.pos.x;
+    effectPos.y = Rand_CenteredFloat(30.0f) + this->actor.world.pos.y;
+    effectPos.z = Rand_CenteredFloat(30.0f) + this->actor.world.pos.z;
+
+    effectAccel.z = 0.0f;
+    effectAccel.y = 0.0f;
+    effectAccel.x = 0.0f;
+
+    effectVel.z = 0.0f;
+    effectVel.y = 0.0f;
+    effectVel.x = 0.0f;
+
+    EffectSsDeadDb_Spawn(globalCtx, &effectPos, &effectVel, &effectAccel, 100, 10, 255, 255, 255, 255, 0, 0, 255, 1, 9,
+                         1);
+}
+
+void EnSkj_SetupWaitInRange(EnSkj* this) {
+    this->textId = 0x10BC;
+
+    EnSkj_ChangeAnim(this, SKJ_ANIM_WAIT);
+    EnSkj_SetupAction(this, SKJ_ACTION_SARIA_SONG_WAIT_IN_RANGE);
+}
+
+void EnSkj_WaitInRange(EnSkj* this, GlobalContext* globalCtx) {
+    Player* player;
+    u16 reactionId;
+
+    player = PLAYER;
+    // When link pulls out the Ocarina center him on the stump
+    // Dev likely forgot to center link as well
```

Comment:
try not to refer to "the devs" directly.
try to phrase the comment like "Link was probably meant to be centered as well" or something

Reply:
and even then, idk if we can really say that. might be worth just not including at all, but im indifferent on it. just seems hard to discern their intention with the info we have

Reply:
My reasoning for mentioning it is because link can be facing away from skull kid and still interact with him normally, which looks unnatural.

Reply:
he spins toward the skullkid tho right? this is just about being centered on the stump positionally

Reply:
yea, he does spin. you could argue that snapping link would look more unnatural than just leaving him where the player put him. so i dunno we can really say

Reply:
oh, i was looking at ocarina minigame, sorry.
so his position is changed, but the angle is not set. Would try to make that more clear in the wording